File: equipo/java/longest_substring_wihtout_repeating_characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int greatestSubstringLength = 0;
        int index = 0;
        int charIndex = 0;
        String currentSubstring = "";
        char currentChar = ' ';
        while(index < s.length() && charIndex < s.length()){
            currentChar = s.charAt(charIndex);
            if(currentSubstring.indexOf(currentChar) > -1){
                index++;
                charIndex = index;
                if(greatestSubstringLength < currentSubstring.length()){
                    greatestSubstringLength = currentSubstring.length();
                }
                currentSubstring = "";
            } else {
                currentSubstring += currentChar;
                charIndex++;
            }
        }

        if(greatestSubstringLength < currentSubstring.length()){
            greatestSubstringLength = currentSubstring.length();
        }
        return greatestSubstringLength;
    }
}

File: equipo/python/longest_substring_wihtout_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longestLength = 0
        currentSubstring = []

        index = 0
        while(True):
            for letter in range(index, len(s)):
                if(s[letter] not in currentSubstring):
                    currentSubstring.append(s[letter])
                else:
                    break
            if(len(currentSubstring) > longestLength):
                longestLength = len(currentSubstring)
            currentSubstring.clear()

            index += 1

            if(index > len(s)):
                break  

        return longestLength
